test/functional/test_framework/test_framework.py

```diff
@@ -85,15 +85,15 @@ def split_network(self):
         stop_nodes(self.nodes)
         self.setup_network(True)
 
-    def sync_all(self):
-        if self.is_network_split:
-            sync_blocks(self.nodes[:2])
-            sync_blocks(self.nodes[2:])
-            sync_mempools(self.nodes[:2])
-            sync_mempools(self.nodes[2:])
-        else:
-            sync_blocks(self.nodes)
-            sync_mempools(self.nodes)
+    def sync_all(self, node_groups=None):
+        if not node_groups:
+            if hasattr(self, "is_network_split") and self.is_network_split:
+                node_groups = [self.nodes[:2], self.nodes[2:]]
+            else:
+                node_groups = [self.nodes]
+
+        [sync_blocks(group) for group in node_groups]
+        [sync_mempools(group) for group in node_groups]
```

Comment:
If you're going to do this, why not just a single for loop?

    for group in node_groups:
        sync_blocks(group)
        sync_mempools(group)

For me, that's a little easier to read.